Add `navigation` and detail screen (wireframe)

alesrazym opened the pull request.

Commit:
FEAT: Add (rfc to) `navigation` graph.

Add `androidx.navigation:navigation-xxx:2.7.5` dependencies, remove duplicate dependencies & update fragment to 1.6.2.
Add `nav-graph.xml`, update activity.
Update `kotlinCompilerExtensionVersion = "1.5.2"`
Add navigation with placeholder destinations.
Add Toolbar to layout, add back arrow drawable, add to navigation.
Add animations as per Android guide.

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListAdapter.kt:
```diff
@@ -7,9 +7,14 @@ import cz.quanti.razym.rocketapp.databinding.RocketListItemBinding
 class RocketListAdapter(
-    private val rockets: List<Rocket>
+    private val rockets: List<Rocket>,
+    private val itemListener: ItemListener
```
je to detail ale tady bych preferoval to predat spis jako lambdu
```suggestion
    private val onItemClick: (Rocket) -> Unit
```

alesrazym replied:
to vypada lip, je to mnohem mene kodu, dik

alesrazym committed:
FEAT: Add API request `SpaceXApi.listRockets()`

Add `mockk` and `kotest-assertions-core`.
Configure `OkHttpClient` with (reasonable) timeout
Handle rockets fetching error in UI
Add simple `RocketsRepositoryImpl` test
Add `RocketListViewModel.uiState` unit test

havlisimo reviewed (approved):
akorat ta enter a exit animace vypada trochu divne (popEnter a popExit jsou ok), ale jinak super
